lib/rucio/web/rest/flaskapi/v1/dids.py

```diff
@@ -1295,172 +1294,288 @@ def generate(vo):
+                    recursive=param_get(parameters, 'recursive', default=False),
+                    vo=vo
+                )
+            except DataIdentifierNotFound as error:
+                return generate_http_error_flask(404, error)
+            except (KeyNotFound, InvalidMetadata, InvalidValueForKey) as error:
+                return generate_http_error_flask(400, error)
+            return 'Created', 201
 
-        return "Created", 201
+        else:
+            meta = param_get(parameters, 'meta')
+            try:
+                set_metadata_bulk(
+                    scope=scope,
+                    name=name,
+                    meta=meta,
+                    issuer=request.environ['issuer'],
+                    recursive=param_get(parameters, 'recursive', default=False),
+                    vo=vo,
+                )
+            except DataIdentifierNotFound as error:
+                return generate_http_error_flask(404, error)
+            except (KeyNotFound, InvalidMetadata, InvalidValueForKey) as error:
+                return generate_http_error_flask(400, error)
+            return "Created", 201
 
-    def delete(self, scope_name):
+    def delete(self, scope_name, key=None):
         """
         ---
         summary: Delete metadata
-        description: "Deletes the specified metadata from the DID."
+        description: |
+          Delete a specific metadata key from a data identifier (DID).
+          This `key` must be provided via the query parameter `?key=...`.
         tags:
           - Data Identifiers
         parameters:
         - name: scope_name
           in: path
-          description: "The scope and the name of the DID."
+          description: "The scope and the name of the DID (e.g., `scope:name`)."
+          required: true
+          style: simple
           schema:
             type: string
-          style: simple
         - name: key
           in: query
-          description: "The key to delete."
+          description: "The metadata key to delete."
+          required: true
+          style: form
           schema:
             type: string
         responses:
           200:
-            description: "OK"
+            description: "OK – the metadata key was successfully removed."
+            content:
+              text/plain:
+                schema:
+                  type: string
+                  enum: [""]
           400:
-            description: "scope_name could not be parsed."
+            description: "Bad Request – invalid scope_name."
           401:
-            description: "Invalid Auth Token"
+            description: "Unauthorized – invalid Auth Token."
           404:
-            description: "DID or key not found"
-          406:
-            description: "Not acceptable"
+            description: >
+              Not found – the specified DID or `key` does not exist, or no `key` query
+              parameter provided.
```

Comment:
404 is fine here but since `key` is a required parameter, I feel like the right error code for 'no `key` query parameter provided' should be 400 or 422 (https://stackoverflow.com/questions/3050518/what-http-status-response-code-should-i-use-if-the-request-is-missing-a-required). I think it's fine to leave it like this too though, reading about it online it seems that 404 could be considered the right response in this use case.